### PR TITLE
Adapt docker recipes to Python 3.11

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Motivation

As we dropped Python 3.7, I updated the docker image recipes to Python 3.11.

## Description of the changes

- Replace Python 3.7 with Python 3.11
- Update torch install command

### CODE REVIEW

**Comment by nabenabe0928** [comment]
@not522 Could you review this PR?

File: kubernetes/mlflow/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,11 @@
 # NOTE(crcrpar): optuna/optuna:py3.7-dev can be used as base instead.
-FROM python:3.7-slim-buster
+FROM python:3.11-slim-buster
```
Comment (not522) [inline kubernetes/mlflow/Dockerfile:2]: `buster` means that this image uses Debian 10. It has been EOL https://www.debian.org/releases/index.en.html, so I think a newer version is preferred. How about `python:3.11-slim`?

Comment (nabenabe0928) [reply kubernetes/mlflow/Dockerfile]: Sorry for the late response, but I applied your suggestion!

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**